### ISSUE TITLE
Jira 9 breaks creating Jira issues

### ISSUE THREAD

**Issue by gforke97** [AtlassianPS/JiraPS#461]
## Description

Get-JiraIssueCreateMetadata fails in Jira Version 9.0 according to a short research its because the used API is deprecreated

## Steps To Reproduce

Run Get-JiraIssueCreateMetadata against a Jira Server with Version 9.0 or higher

## Expected behavior

Get the Meta Data needed to create an Issue

## Your Environment

Jira 9.0

Get-Module JiraPS -ListAvailable | Select Name, Version

Name   Version
----   -------
JiraPS 2.14.6

$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

## Possible Solution

https://bitbucket.org/atlassianlabs/atlascode/issues/753/cannot-open-issue-create-issue-in-vscode

**Comment by gforke97** [comment]
Jira Server not cloud btw

**Comment by rjl61** [comment]
Same for 9.1.  I also discovered that the API call used by Get-JiraIssueCreateMetadata was not only deprecated but missing entirely.  I was already making some local changes to use the new API call when I saw your post.  The workaround does work after also making changes to the ConvertTo-JiraCreateMetaField for the different structure.

**Comment by saeid-adz** [comment]
I have the same problem! @rjl61 would you please share what changes have you made to solve the problem?
It's pretty urgent for me! 
Thanks!

**Comment by jcrivas1971** [comment]
Is there any chance that we will see a solution for this issue?
@rjl61: Could you please provide your solution?
Thanks.

**Comment by spicy** [comment]
@jcrivas1971 @saeid-adz This is my solution. I have not significantly tested this but it seems to work for my purposes. https://github.com/spicy/JiraPS/commit/f2934f22c67d7afe1f678279d198206448ecba72

**Comment by saeid-adz** [comment]
@jcrivas1971 Thanks mate! Worked for me too! I have several types of connection and automation based on this module and I can confirm that the issues have been solved by using your solution! 
Thanks again!

**Comment by jcrivas1971** [comment]
@saeid-adz: I am sorry but I guess you have to thank @spicy.
@spicy: I will try the solution next week. Thanks in advance for your work.

**Comment by saeid-adz** [comment]
@jcrivas1971 Thanks mate, I selected the wrong userId! edited now :)

**Comment by jcrivas1971** [comment]
@spicy: It took a while but I was able to test your fix today. It works perfectly!!
 Thanks a lot.

**Comment by mwoffenden** [comment]
I need to integrate this fix into my local WindowsPowerShell\Modules\JiraPS\2.14.6\JiraPS.psm1, what's the right way to do this?

**Comment by spicy** [comment]
> @spicy I need to integrate this fix **today** into my local WindowsPowerShell\Modules\JiraPS\2.14.6\JiraPS.psm1, what's the right way to do this?  Or can we somehow get a fixed module?

I don't think this project is being supported since my merge request was not accepted yet. You can just copy my code and modify the function if it's visible. Otherwise you can put the new function in a new ps1 to call instead. Hope this helps

**Comment by mwoffenden** [comment]
> You can just copy my code and modify the function if it's visible. Otherwise you can put the new function in a new ps1 to call instead.

@spicy Thanks - I did try that locally, made the edits to "function Get-JiraIssueCreateMetadata" but the error still shows up, so I must be doing something wrong.  We (along with many others) very much need this fixed.  

Any idea how to build an updated module using your fixed code?

**Comment by mwoffenden** [comment]
Can anyone help with this?

**Comment by jcrivas1971** [comment]
@mwoffenden:  spicy forked the original project and provided a fixed version. Instead of a local edit you should probably try to use his version or just the corrected file? You could download the whole project as a zip file and either copy the whole directory or just the above mentioned file.  Just go to spicy/JIRAPS. I hope this helps.

**Comment by mwoffenden** [comment]
@jcrivas1971 Yes, we finally did download the code and made the edits locally, this solution seems to be working for us.  However I can imagine that the ideal is to have the master JiraPS brought up to date so that people downloading from the Powerhsell gallery can have a working version.

**Comment by eXpl0it3r** [comment]
So what's the status with proper Jira 9 support, @lipkau?

**Comment by claudiospizzi** [comment]
Same issue here. What I can see, this is related to this change:
https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

We've created a workaround for our use case, and we use this internally. Not sure if this covers all cases and if it is a complete workaround. Anyway, if someone finds that helpful:
https://github.com/arcadech/JiraPS/commit/5d2805357475c4564ef013d4a36e497d23f68595

**Comment by ChefkeGremmen** [comment]
@claudiospizzi ; thank you so much for this. I hope the maintainers find room to get the CICD pipeline up and running again so your pull-request can get processed. Your change fixed our situation.

**Comment by bradyclifford** [comment]
Seems the alpha pre-release `2.15.0-alpha1` of JiraPS fixes the issue.  I am able to use the `New-JiraIssue` method without any issues.  What can we do to get that pre-release promoted.

**Comment by mwoffenden** [comment]
Thanks @bradyclifford!  Would you post a link to where I can find this pre-release?

**Comment by bradyclifford** [comment]
I am not sure if it applies to Jira Cloud.  I only tried the pre-release with the on-prem version of Jira.  The version is listed in this repo's releases:  https://github.com/AtlassianPS/JiraPS/releases/tag/v2.15.0-alpha1

**Comment by lipkau** [comment]
Fixed in v2.15 (#488) — `Get-JiraIssueCreateMetadata` migrated to the modern `/rest/api/2/issue/createmeta/{project}/issuetypes/{type}` endpoint, which is the supported one on Jira 9+. `Update-Module JiraPS`. Closing.
